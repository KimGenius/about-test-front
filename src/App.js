import './App.css'
import {Button, Table, TableBody, TableCell, TableHead, TableRow, TextField} from "@mui/material"
import {useEffect, useState} from "react"
import axios from "axios"
import dayjs from "dayjs"

function App() {
  let [newMember, setNewMember] = useState('')
  let [targetMember, setTargetMember] = useState('')
  let [babList, setBabList] = useState([])
  useEffect(() => {
    getBabHistory()
  }, [])

  async function getBabHistory() {
    const url = "http://localhost:5000/bab"
    const { data } = await axios.get(url)
    setBabList(data)
  }

  async function babWithNewMember() {
    const url = "http://localhost:5000/bab"
    try {
      const result = await axios.post(url, {
        newMember, targetMember
      })
      console.log(result)
      getBabHistory()
    } catch (e) {
      alert(e.response.data.message)
    }
  }
  return (
    <div className="App">
      <TextField id="standard-basic" label="신규 입사자" value={newMember} onChange={e => setNewMember(e.target.value)}/>
      <TextField id="standard-basic" label="출근자 리스트" value={targetMember} onChange={e => setTargetMember(e.target.value)} />
      <Button onClick={() => babWithNewMember()}>밥 머거</Button>
      <Table sx={{ minWidth: 650 }} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>신규입사자</TableCell>
            <TableCell align="right">같이 밥 먹은 팀원</TableCell>
            <TableCell align="right">밥 먹은 날</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {babList.map((bab) => (
            <TableRow
              key={bab._id}
              sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
              <TableCell component="th" scope="row">
                {bab.newMember}
              </TableCell>
              <TableCell align="right">{bab.targetMember}</TableCell>
              <TableCell align="right">{dayjs(bab.createdAt).format('YYYY-MM-DD HH:mm:ss')}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </div>
  )
}

export default App
