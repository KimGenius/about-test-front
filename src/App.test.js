import {render, screen, waitFor} from '@testing-library/react';
import App from './App';
import axios from 'axios';

jest.mock('axios');

test('renders learn react link', async () => {
  axios.get.mockResolvedValue({
    data: [
      {
        "idx": 1,
        "newMember": "a",
        "targetMember": "c,d,f,ef,,g,t,h,hj",
        "createdAt": "2021-10-27T05:18:27.000Z",
        "resultMember": "hj,g"
      }
    ],
  });
  const { getByTestId, asFragment }= render(<App />)
  const listNode = await waitFor(() => getByTestId("list"))
  expect(listNode.children).toHaveLength(1);
  expect(asFragment()).toMatchSnapshot();
  let targetEle = screen.getByText(/같이 밥 먹은 팀원/i);
  expect(targetEle).toBeInTheDocument();
  targetEle = screen.getByText(/신규입사자/i);
  expect(targetEle).toBeInTheDocument();
  // targetEle = screen.getByText(/퇴근한 팀원들/i);
  // expect(targetEle).toBeInTheDocument();
  targetEle = screen.getByText(/밥 머거/i);
  expect(targetEle).toBeInTheDocument();
});
