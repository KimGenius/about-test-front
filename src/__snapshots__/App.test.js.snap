// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="standard-basic"
        id="standard-basic-label"
      >
        신규 입사자
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
          id="standard-basic"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-dhh0px"
          >
            <span>
              신규 입사자
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      style="width: 60vw;"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="standard-basic"
        id="standard-basic-label"
      >
        출근자 리스트
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
          id="standard-basic"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-dhh0px"
          >
            <span>
              출근자 리스트
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      밥 머거
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <table
      aria-label="simple table"
      class="MuiTable-root css-quj9j6-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            신규입사자
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
            scope="col"
          >
            같이 밥 먹은 팀원
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
            scope="col"
          >
            출근한 팀원들
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
            scope="col"
          >
            밥 먹은 날
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        data-testid="list"
      >
        <tr
          class="MuiTableRow-root css-34nofg-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            a
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            hj,g
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            c,d,f,ef,,g,t,h,hj
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            2021-10-27 14:18:27
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
